org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/resources/servicetemplates/ServiceTemplateResource.java

```diff
@@ -226,17 +228,17 @@ public Response injectNodeTemplates(InjectorReplaceData injectorReplaceData) thr
 			}
 		}
 		TTopologyTemplate tTopologyTemplate = splitting.injectNodeTemplates(this.getServiceTemplate().getTopologyTemplate(), injectorReplaceData.injections);
+		this.getServiceTemplate().setTopologyTemplate(null);
 		this.getServiceTemplate().setTopologyTemplate(tTopologyTemplate);
-		ServiceTemplateId injectedServiceTemplateId = new ServiceTemplateId(id.getNamespace().getDecoded(), id.getXmlId().getDecoded() + "-injected", false);
-		Repository.INSTANCE.forceDelete(injectedServiceTemplateId);
-		Repository.INSTANCE.flagAsExisting(injectedServiceTemplateId);
-		ServiceTemplateResource splitServiceTempateResource = (ServiceTemplateResource) AbstractComponentsResource.getComponentInstaceResource(injectedServiceTemplateId);
-
-		splitServiceTempateResource.getServiceTemplate().setTopologyTemplate(tTopologyTemplate);
 		LOGGER.debug("Persisting...");
-		splitServiceTempateResource.persist();
+		this.persist();
 		LOGGER.debug("Persisted.");
-		return Utils.getXML(TTopologyTemplate.class, tTopologyTemplate);
+
+		//No renaming of the Service Template allowed because of the plans
+
+		URI url = uriInfo.getBaseUri().resolve(Utils.getAbsoluteURL(id));
+		LOGGER.debug("URI of the old and new ServiceTemplate: " + url.toString());
```

Comment:
Can you use `LOGGER.debug("URI of the old and new service template {}", url.toString());` please?

Reason: http://stackoverflow.com/a/10596390/873282